crates/control/src/role_assignment.rs

```diff
@@ -372,289 +285,320 @@ impl RoleAssignment {
+        cycle_start_time
+            .duration_since(team_ball.last_seen)
+            .expect("time ran backwards")
+            <= striker_trusts_team_ball_duration
+    };
 
-        (Role::Loser, None, Some(spl_message)) => match &spl_message.ball_position {
-            None => (Role::Loser, false, None), //edge-case, a striker (which should not exist) lost the ball
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
+    match (current_role, detected_own_ball, event) {
+        // Striker lost Ball
+        (Role::Striker, false, Event::None | Event::Loser) => match team_ball {
+            Some(team_ball) if striker_trusts_team_ball(team_ball) => Role::Striker,
+            _ => Role::Loser,
         },
 
-        //Loser found ball and becomes Striker
-        (Role::Loser, Some(..), None) => (
-            Role::Striker,
-            true,
-            team_ball_from_seen_ball(detected_own_ball, current_pose, cycle_start_time),
+        (_other_role, _, Event::Striker(striker_event)) => claim_striker_or_other_role(
+            striker_event,
+            time_to_reach_kick_position,
+            player_number,
+            filtered_game_controller_state,
+            optional_roles,
         ),
 
-        // Edge-case, Loser found Ball at the same time as receiving a message
-        (Role::Loser, Some(..), Some(spl_message)) => match &spl_message.ball_position {
-            None => {
-                // another Striker became Loser, so we claim striker since we see a ball
-                (
-                    Role::Striker,
-                    true,
-                    team_ball_from_seen_ball(detected_own_ball, current_pose, cycle_start_time),
-                )
-            }
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
-        },
-
-        //Searcher remains Searcher
-        (Role::Searcher, None, None) => (Role::Searcher, false, team_ball),
-
-        (Role::Searcher, None, Some(spl_message)) => match &spl_message.ball_position {
-            None => (Role::Searcher, false, team_ball), //edge-case, a striker (which should not exist) lost the ball
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
+        // Striker remains Striker, sends message after timeout
```

Comment:
why this comment part about "sends message after timeout"?

Reply:
Removed